tests/unit/handle-changes-test.js

```diff
@@ -0,0 +1,44 @@
+var simple = require('simple-mock')
+var test = require('tape').test
+
+var handleChanges = require('../../lib/helpers/handle-changes')
+
+test('handleChanges changes during bootstrapping', function (t) {
+  t.plan(2)
+
+  var change = {
+    doc: {
+      foo: 'bar'
+    },
+    id: 'id1234'
+  }
+  var changesSpy = {
+    on: simple.spy().callFn(function (changeEvent, callback) {
+      callback(change)
+    })
+  }
+
+  var state = {
+    db: {
+      allDocs: simple.spy().resolveWith({ rows: [] }),
+      changes: simple.spy().returnWith(changesSpy)
+    },
+    emitter: {
+      emit: simple.spy()
+    }
+  }
+
+  handleChanges(state)
+
+  Promise.resolve(state.bootstrap)
```

Comment:
> state.bootstrap is undefined

Actually, `state.bootstrap` is a Promise, per the [assignment here](https://github.com/hoodiehq/hoodie-store-client/blob/56a435127eda9dbe239a00b5e38ee51ce7c75ca7/lib/helpers/handle-changes.js#L12-L22).  What happens is, `handleChange` runs synchronously and stores the pending Promise from `state.db.allDocs().then()` call into `state.bootstrap`, so `state.bootstrap` is actually a Promise.

But I see your point -- there's nothing to prevent resolving the `.then()` call chain after the `state.db.allDocs()`, which we want to control the flow of.  I'll work on refactoring that. :+1: